common/sc_popen.cpp

```diff
@@ -31,10 +31,19 @@
 FILE *
 sc_popen(const char *command, pid_t *pidp, const char *type)
 {
-	FILE *iop;
-	int pdes[2], pid, twoway;
 	char *argv[4];
+	argv[0] = (char *)"sh";
```

Comment:
But `sh.c_str()` returns `const char*`, a pointer to constant char, this will not work since argv should be `char* const x[]`. So .data() will return a pointer to constant pointer to *constant char*, while argv should be pointer to constant pointer to *char*.  So it seems, string can't be used, to do it correctly I would need to create a char[] of the correct size and copy the string there ? The use of std::array is fine I think.

Weird why execve asks in the filename for pointer to constant char, but with argv asks for array of constant pointer to char...

The mix of const, pointers and [] is a bit mind boggling. I needed a translator to make sense of this ... http://cdecl.o